### PR TITLE
Document Auto-Parameterization.

### PR DESCRIPTION
**Pull request by sherfert**

### CODE REVIEW

**Review by tobias-johansson — APPROVED** [review]
LGTM

File: cypher/cypher-docs/src/docs/dev/syntax/parameters.asciidoc [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 
 * <<cypher-parameters-introduction,Introduction>>
+* <<cypher-parameters-auto-parameterization,Auto-Parameterization>>
```
Comment (davidoliverSP2) [inline cypher/cypher-docs/src/docs/dev/syntax/parameters.asciidoc]:
```suggestion
* <<cypher-parameters-auto-parameterization,Auto-parameterization>>
```

File: cypher/cypher-docs/src/docs/dev/syntax/parameters.asciidoc [hunk 2]
```diff
@@ -56,6 +57,20 @@ However, it is still possible to use it, with warnings, if you prefix the query
 See <<cypher-compatibility, Cypher Compatibility>> for further information.
 ====
 
+[[cypher-parameters-auto-parameterization]]
+== Auto-Parameterization
```
Comment (davidoliverSP2) [inline cypher/cypher-docs/src/docs/dev/syntax/parameters.asciidoc]:
```suggestion
== Auto-parameterization
```

**Review by davidoliverSP2 — APPROVED** [review]
Thanks @sherfert 👍

**Pull request merged**